File: code_samples/swift/data_structures_linked_list.swift
import Foundation

/// Data Structures: Linked List
/// AI/ML Training Sample

class LinkedList {
    var data: String
    
    init() {
        self.data = ""
    }
    
    func process(_ input: String) {
        self.data = input
    }
    
    func getData() -> String {
        return self.data
    }
    
    func validate() -> Bool {
        return !self.data.isEmpty
    }
}

// Example usage
let instance = LinkedList()
instance.process("example")
print("Data: \(instance.getData())")
print("Valid: \(instance.validate())")

File: code_samples/dart/data_structures_linked_list.dart
// Data Structures: Linked List
// AI/ML Training Sample

class LinkedList {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = LinkedList();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
